tests/integration/cgroups.bats

```diff
@@ -176,3 +178,37 @@ function setup() {
     [ "$status" -eq 0 ]
 #
 }
+
+@test "runc run (cgroup v1 + unified resources should fail)" {
+    requires root cgroups_v1
+
+    set_cgroups_path "$BUSYBOX_BUNDLE"
+    set_resources_limit "$BUSYBOX_BUNDLE"
+    update_config '.linux.resources.unified |= {"memory.min": "131072"}' "$BUSYBOX_BUNDLE"
+
+    runc run -d --console-socket "$CONSOLE_SOCKET" test_cgroups_unified
+    [ "$status" -ne 0 ]
+    [[ "$output" == *'invalid configuration'* ]]
+}
+
+@test "runc run (cgroup v2 + unified resources)" {
+    requires root cgroups_v2
+
+    set_cgroups_path "$BUSYBOX_BUNDLE"
+    update_config ' .linux.resources.memory |= {"limit": 33554432}
+                  | .linux.resources.memorySwap |= {"limit": 33554432}
+                  | .linux.resources.unified |=
+                      {"memory.min": "131072", "memory.max": "10485760" }' \
```

Comment:
We should also test:
* `pids.max` (which doesn't conflict v1-compat config)
* `non.existent.value`
* `../../../etc/shadow`

Reply:
I am already testing the last two in the libcontainer/integration tests. Not sure how adding these cases to bats will add any value.

Not sure what you mean by "v1-compat config". The runtime-spec says 'resources.unified' can't be used on v1, and there's a code to enforce it (checking if "unified" is set, doesn't look into individual parameters). Do you mean you want to test this works if we set "pids.max"?

Reply:
All the current fields in `.linux.resources.unified` overrides the v1 config (`.linux.resources.{memory,memorySwap}`).

We should ensure that we can set unified fields that do not override v1.

Reply:
There is a code that rejects config if `.linux.resources.unified` set for cgroupv1, here:
https://github.com/opencontainers/runc/pull/2584/commits/b14556f2678eb12cbacdc12ae3ddf6dc952e3fbe#diff-252ac01f71681e9ed8e7ce0571cdb6acR349-R351
and here:
https://github.com/opencontainers/runc/pull/2584/commits/b14556f2678eb12cbacdc12ae3ddf6dc952e3fbe#diff-bf0fd9d73921163a3f447b24a8187ee7

I check that this code works in two tests, here:
https://github.com/opencontainers/runc/pull/2584/commits/b14556f2678eb12cbacdc12ae3ddf6dc952e3fbe#diff-bd87cca3665e082adb4ba55ee22e82b5
and here:
https://github.com/opencontainers/runc/pull/2584/commits/d0397b925803e3cc78e79ec7ef1da622529595c1#diff-1be821a9fd1920ed1cd1090d8a6bea39R182

The rejection code does not look into what's set under the `unified`, so it does not make sense to add some specific parameters in there -- anything is rejected.